### ISSUE TITLE
Roadmap to v1.0

### ISSUE THREAD

**Issue by mpgirro** [mpgirro/stalla#50]
The question is still open what's on the roadmap before we push a v1.0 release? The API of what we currently have implemented should be definitely stable at this point. For that, I'm currently looking again at the specifications (and recognised #43 and #44 for now).

- [x] #34 Parse episode duration value
- [x] #43 iTunes category restrictions
- [x] #44 Google Play category restrictions
- [x] #52 Add Detekt (and fix issues)
- [x] #60 Custom MediaType

Current state on other issues:

- #35 Unsigned numeric values: This should really be completed before v1.0, but for reasons stated in the item, it is currently on hold.
- #36 Orchid-based website: Currently there is a PoC in branch [orchid](/mpgirro/stalla/tree/orchid). Site is generating, but Dokka API is not satisfyingly integrated yet. A website however is a nice to have and can always be provided later on and is therefore not a requirement to get v1.0 out in my opinion. A decent documentation in the GitHub README would be more important first. 
- Additional namespaces like #28 Dublin Core, #29 Spotify, #30 Media RSS, #49 Podcastindex phase 2 are all additions and well suited for v1.x releases. Hence no showstopper for a release.

What are your thoughts on this @rock3r?

**Comment by rock3r** [comment]
You are indeed correct. I think we're good to go, apart for finalising the changes we've talked about for Play, and doing the duration parsing. I wouldn't wait on the unsigned numbers stuff — we can delegate the validation of the numbers to a validator in the future, and once Kotlin 1.5 is out, we can consider again what's the interop looking like and maybe slate it for 2.0

**Comment by rock3r** [comment]
I was wondering, would it be worth it to add Detekt to the project so we can fix any issues it finds before going gold?

**Comment by mpgirro** [comment]
I've no experience with Detekt yet, but yes, could be worth it. The more we find now the better.

**Comment by rock3r** [comment]
Have created an issue, and updated this issue's description to match

**Comment by rock3r** [comment]
Note for the future, when we get to publish to Central, we can use https://github.com/gradle-nexus/publish-plugin

**Comment by rock3r** [comment]
I've added issue #55 to set up binary compatibility checks for APIs, which is quite important given we use data classes and it's very easy to [break binary compatibility](https://jakewharton.com/public-api-challenges-in-kotlin/) with those

**Comment by rock3r** [comment]
Also created #56 to track my earlier comment

**Comment by mpgirro** [comment]
All concerns here (and some more 😅) have been addresses now.

**Comment by rock3r** [comment]
Awesomeeeee  🎉
